service/src/test/org/apache/hive/service/cli/TestCLIServiceConnectionLimits.java

```diff
@@ -345,6 +347,83 @@ public void testConnectionForwardedIpAddresses() throws HiveSQLException {
     }
   }
 
+  @Test
+  public void testConnectionLimitPerUserWhenSessionCreationFails() throws HiveSQLException {
+    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_LIMIT_CONNECTIONS_PER_USER, 2);
+    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_LIMIT_CONNECTIONS_PER_IPADDRESS, 0);
+    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_LIMIT_CONNECTIONS_PER_USER_IPADDRESS, 0);
+    CLIService service = getService(conf);
+    HashMap<String, String> sessionConf = new HashMap<>(1);
+    // modifying this property at runtime is not allowed, it will throw exception and session will not be created
+    sessionConf.put("hive.in.test", "true");
```

Comment:
Could you please use another restricted config? IMHO the presence of the `hive.in.test` setting is wrong, as it indicates that test-specific code is being shipped to the production.